src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -1207,26 +1235,47 @@ private static Interop.Http.CurlSeekResult CurlSeekCallback(IntPtr context, long
                     CurlHandler.EventSourceTrace("Offset: {0}, Origin: {1}", offset, origin, 0, easy: easy);
                     try
                     {
-                        // If libcul is requesting we seek back to the beginning and if the request
-                        // content stream is in a position to reset itself, reset and let libcurl
-                        // know we did the seek; otherwise, let it know we can't seek.
-                        if (offset == 0 && origin == (int)SeekOrigin.Begin && 
-                            easy._requestContentStream != null && easy._requestContentStream.TryReset())
+                        // If we don't have a stream yet, we can't seek.
+                        if (easy._requestContentStream == null)
                         {
-                            // Dump any state associated with the old stream's position
-                            easy._sendTransferState?.SetTaskOffsetCount(null, 0, 0);
-
-                            // Restart the transfer
-                            easy._requestContentStream.Run();
+                            CurlHandler.EventSourceTrace("No request stream exists yet. Can't seek", easy: easy);
+                            return Interop.Http.CurlSeekResult.CURL_SEEKFUNC_CANTSEEK;
```

Comment:
Hmm, interesting point.  I don't think it'll do that.  And at this point with most folks signed off, I'd rather not add such additional logic.  We can certainly add that easily if we ever see it happening.